code/edgeconsent/src/main/java/com/adobe/marketing/mobile/edge/consent/ConsentExtension.java

```diff
@@ -30,7 +30,9 @@
 class ConsentExtension extends Extension {
 
 	private static final String LOG_SOURCE = "ConsentExtension";
-	private final ConsentManager consentManager;
+
+	// package private vars
+	ConsentManager consentManager;
```

Comment:
I suggest injecting this variable via constructor and retain its final nature while making it testable. That way you can mock this directly in your test as well as eliminate the whitebox mock.

```
        protected ConsentExtension(final ExtensionApi extensionApi) {
		this(
			extensionApi,
			new ConstentManager(ServiceProvider
				.getInstance()
				.getDataStoreService()
				.getNamedCollection(ConsentConstants.DataStoreKey.DATASTORE_NAME))
		);
	}

	protected ConsentExtension(final ExtensionApi extensionApi, final ConsentManager consentManager) {
		super(extensionApi);
		consentManager = this .consentManager;
	}
```